compiler/src/iree/compiler/Dialect/HAL/Target/VulkanSPIRV/VulkanSPIRVTarget.cpp

```diff
@@ -48,6 +48,11 @@ VulkanSPIRVTargetOptions getVulkanSPIRVTargetOptionsFromFlags() {
       "iree-vulkan-target-triple", llvm::cl::desc("Vulkan target triple"),
       llvm::cl::init("unknown-unknown-unknown"));
 
+  static llvm::cl::opt<int> clVulkanIndexingBits(
```

Comment:
Actually I think we should do this as a SPIR-V level option, instead of Vulkan level. Meaning, this option should be placed in `SPIRV/Passes.cpp` and named as like `iree-spirv-index-bitwidth`. The reasons are that 1) it conceptually is a SPIR-V switch--we are treating indices as 32 or 64 bits within SPIR-V; 2) we can share the same option across various SPIR-V consumer APIs, so we don't need to create separate ones for Vulkan, OpenCL, Metal, WebGPU, etc.

Reply:
Makes sense. Initially it was only in place for OpenCL, (and was flipped on with a flag for physical64 addressing) hence why I set the option like this, but there's no reason to do that for other consumers.

Reply:
Better to take it out too. I don't see a use case to unconditionally turn it on/off for a certain client API yet.